server/resp/src/main/java/org/infinispan/server/resp/RespServer.java

```diff
@@ -48,7 +48,7 @@ protected void startInternal() {
          } else {
             if (cacheManager.getCacheManagerConfiguration().isClustered()) { // We are running in clustered mode
                builder.clustering().cacheMode(CacheMode.REPL_SYNC);
-               builder.clustering().hash().hashFunction(CRC16.getInstance());
+               builder.clustering().hash().keyPartitioner(new CRC16HashFunctionPartitioner()).numSegments(256);
```

Comment:
is `.numSegments(256)` required?

Reply:
Not really. I just added it because it was the original value. I'll remove it when fixing the check style.

But this also reminds me, [on their specification](https://redis.io/docs/reference/cluster-spec/#key-distribution-model), they have this huge number for _slots_ of `16384`. I can't tell if we should be using that instead? 

/cc @wburns

Reply:
Probably, it shouldn't hurt. Do we have any test to check if the RESP client sends the request to the correct server?

Reply:
Updated. Right now, for RESP, we have just a couple of tests https://github.com/infinispan/infinispan/blob/main/server/resp/src/test/java/org/infinispan/server/resp/ClusteredOperationsTest.java#L9 and https://github.com/infinispan/infinispan/blob/main/server/resp/src/test/java/org/infinispan/server/resp/RespTwoNodeTest.java#L26

Reply:
As far as I can tell, none of those tests check if the expected server received the request. And since it is only a 2 nodes cluster, all servers are the owners which read from locally even if the CRC16 is incorrectly implemented  (I'm not telling that it is, just want to check that it isn't)

Reply:
I created https://issues.redhat.com/browse/ISPN-14894 so we can tackle that. Especially since we need to increase our tests with multiple nodes.